cpu/nrf5x_common/periph/timer.c

```diff
@@ -113,6 +117,55 @@ int timer_set_absolute(tim_t tim, int chan, unsigned int value)
     return 0;
 }
 
+int timer_set(tim_t tim, int chan, unsigned int timeout)
+{
+    static const uint32_t max_mask[] = {
+        [TIMER_BITMODE_BITMODE_08Bit] = 0x000000ff,
+        [TIMER_BITMODE_BITMODE_16Bit] = 0x0000ffff,
+        [TIMER_BITMODE_BITMODE_24Bit] = 0x00ffffff,
+        [TIMER_BITMODE_BITMODE_32Bit] = 0xffffffff,
+    };
+    /* see if channel is valid */
+    if (chan >= timer_config[tim].channels) {
+        return -1;
+    }
+
+    unsigned value = timer_read(tim) + timeout;
+
+    ctx[tim].flags |= (1 << chan);
```

Comment:
```suggestion
    ctx[tim].flags |= (1 << chan);
    ctx[tim].is_periodic &= ~(1 << chan);
```

this is also missing from `timer_set_absolute()`

Reply:
Fixed. Also fixed for `timer_set_absolute()` in a separate commit. Finally, this is a read-modify-write operation that should be atomic when allowing two threads to share one timer (but each thread using a distinct set of timer channels). I fixed that in a third commit.